workspaces/ballerina/ballerina-side-panel/src/components/editors/MultiModeExpressionEditor/ChipExpressionEditor/ChipExpressionBaseComponent.tsx

```diff
@@ -328,11 +357,21 @@ export const ChipExpressionBaseComponent = (props: ChipExpressionBaseComponentPr
         );
     }, [isCompletionsOpen, selectedCompletionItem, filteredCompletions]);
 
+    const handleHelperKeyDown = useCallback((e: React.KeyboardEvent) => {
+        if (e.key === 'Escape') {
+            e.preventDefault();
+            setIsHelperPaneOpen(false);
+        }
+    }, [isCompletionsOpen, selectedCompletionItem, filteredCompletions]);
```

Comment:
The dependency array includes variables that aren't used in the function, which is misleading. Does this callback wrapper really needed?